### PR TITLE
Report racktables results and issues for the user

### PR DESCRIPTION
**Pull request by nicksinger**

### CODE REVIEW

**Review by Martchus — APPROVED** [review]
I guess it is fine although it would probably be useful to log also the response and possibly even the route the script tried to query.

File: racktables/find_pingable_machines.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 python3 get_unused_machines.py | while read fqdn; do
     ping -c1 -W1 "${fqdn}" &> /dev/null && echo "${fqdn} up"
-done
+done && exit 1 || exit 0
```
Comment (okurz) [inline racktables/find_pingable_machines.sh:4]: If you want to turn around the exit code you should be able to do that with

```
! python3 … ; do
…
done
```

Comment (nicksinger) [reply racktables/find_pingable_machines.sh:4]: your suggestion looks good. However while testing I realized that without a temporary variable to store the state, I will most likely not archive what I want with this change (fail if _any_ host is reachable, not just fail if the last one is reachable)

Comment (okurz) [reply racktables/find_pingable_machines.sh:4]: ok, nevermind. Just keep as is then I guess

File: racktables/racktables.py [hunk 2]
```diff
@@ -18,6 +18,11 @@ def search(self, search_payload={}):
             join(self.url, "index.php"),
             params="&".join("%s=%s" % (k, v) for k, v in search_payload.items()),
         )
+        status = req.status_code
+        if status == 401:
+          raise Exception("Racktables returned 401 Unauthorized. Are your credentials correct?")
+        elif status >= 300:
+          raise Exception(f"Racktables returned statuscode {status} for a search request. Manual investigation needed.")
```
Comment (kalikiana) [inline racktables/racktables.py:25]: Might be worth including the search terms if we can? To make it easy to reproduce

Comment (nicksinger) [reply racktables/racktables.py]: I tried to incorporate your feedback. Are you fine with that?

File: racktables/find_pingable_machines.sh (hunk 1 above)
Comment (nicksinger) [reply racktables/find_pingable_machines.sh]: got pretty ugly now. If you know of a better way please let me know :)

**Review by okurz — APPROVED** [review]

File: racktables/find_pingable_machines.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+found_machines=0
+while read fqdn; do
```
Comment (baierjan) [inline racktables/find_pingable_machines.sh:3]: shellcheck would probably suggests

```suggestion
while read -r fqdn; do
```

File: racktables/find_pingable_machines.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+    ping -c1 -W1 "${fqdn}" &> /dev/null && echo "${fqdn} up" && found_machines=$(($found_machines + 1))
```
Comment (baierjan) [inline racktables/find_pingable_machines.sh:4]: You are in "math mode" here
```suggestion
    ping -c1 -W1 "${fqdn}" &> /dev/null && echo "${fqdn} up" && found_machines=$((found_machines + 1))
```

File: racktables/find_pingable_machines.sh [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+done <<< $(python get_unused_machines.py)
+exit $(($found_machines > 0))
```
Comment (baierjan) [inline racktables/find_pingable_machines.sh:6]:
```suggestion
exit $((found_machines > 0))
```

File: racktables/find_pingable_machines.sh (hunk 3 above)
Comment (nicksinger) [reply racktables/find_pingable_machines.sh]: might make sense if machine names contain funny characters :)

**Review by Martchus — APPROVED** [review]

**Pull request merged**